### ISSUE TITLE
File uploaded to S3 is smaller that file selected on Device/Client (Expo)

### ISSUE THREAD

**Issue by quocluongha** [aws-amplify/amplify-js#11549]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Storage

### Amplify Categories

storage

### Environment information

<details>

```
# Put output below this line
 System:
    OS: macOS 13.4.1
    CPU: (10) arm64 Apple M2 Pro
    Memory: 87.17 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.0 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/Workspace/Images-Social-Mobile/node_modules/.bin/yarn
    npm: 9.6.7 - /opt/homebrew/bin/npm
    Watchman: 2023.06.12.00 - /opt/homebrew/bin/watchman
  Browsers:
    Brave Browser: 114.1.52.126
    Safari: 16.5.1
  npmPackages:
    @apollo/client: ^3.7.1 => 3.7.16 
    @apollo/client/cache:  undefined ()
    @apollo/client/core:  undefined ()
    @apollo/client/errors:  undefined ()
    @apollo/client/link/batch:  undefined ()
    @apollo/client/link/batch-http:  undefined ()
    @apollo/client/link/context:  undefined ()
    @apollo/client/link/core:  undefined ()
    @apollo/client/link/error:  undefined ()
    @apollo/client/link/http:  undefined ()
    @apollo/client/link/persisted-queries:  undefined ()
    @apollo/client/link/retry:  undefined ()
    @apollo/client/link/schema:  undefined ()
    @apollo/client/link/subscriptions:  undefined ()
    @apollo/client/link/utils:  undefined ()
    @apollo/client/link/ws:  undefined ()
    @apollo/client/react:  undefined ()
    @apollo/client/react/components:  undefined ()
    @apollo/client/react/context:  undefined ()
    @apollo/client/react/hoc:  undefined ()
    @apollo/client/react/hooks:  undefined ()
    @apollo/client/react/parser:  undefined ()
    @apollo/client/react/ssr:  undefined ()
    @apollo/client/testing:  undefined ()
    @apollo/client/testing/core:  undefined ()
    @apollo/client/utilities:  undefined ()
    @apollo/client/utilities/globals:  undefined ()
    @babel/core: ^7.20.0 => 7.20.12 (7.22.5, 7.22.1)
    @babel/preset-env: ^7.20.0 => 7.20.2 
    @babel/runtime: ^7.20.0 => 7.20.7 (7.22.5, 7.22.3, 7.21.0)
    @gorhom/bottom-sheet: ^4.4.7 => 4.4.7 
    @graphql-codegen/cli: ^4.0.1 => 4.0.1 
    @graphql-codegen/typescript: ^4.0.0 => 4.0.1 
    @graphql-codegen/typescript-apollo-client-helpers: ^2.2.6 => 2.2.6 
    @graphql-codegen/typescript-operations: ^4.0.0 => 4.0.1 
    @graphql-codegen/typescript-react-apollo: ^3.3.7 => 3.3.7 
    @graphql-codegen/typescript-resolvers: ^4.0.1 => 4.0.1 
    @hookform/resolvers: ^3.1.0 => 3.1.1 
    @hookform/resolvers/ajv:  1.0.0 
    @hookform/resolvers/arktype:  1.0.0 
    @hookform/resolvers/class-validator:  1.0.0 
    @hookform/resolvers/computed-types:  1.0.0 
    @hookform/resolvers/io-ts:  1.0.0 
    @hookform/resolvers/joi:  1.0.0 
    @hookform/resolvers/nope:  1.0.0 
    @hookform/resolvers/superstruct:  1.0.0 
    @hookform/resolvers/typanion:  1.0.0 
    @hookform/resolvers/typebox:  1.0.0 
    @hookform/resolvers/vest:  1.0.0 
    @hookform/resolvers/yup:  1.0.0 
    @hookform/resolvers/zod:  1.0.0 
    @openspacelabs/react-native-zoomable-view: ^2.1.1 => 2.1.1 
    @react-native-async-storage/async-storage: ^1.17.11 => 1.18.2 
    @react-native-community/eslint-config: ^3.2.0 => 3.2.0 
    @react-native-community/netinfo: ^9.3.10 => 9.3.10 
    @react-native-firebase/app: ^17.5.0 => 17.5.0 
    @react-navigation/bottom-tabs: ^6.4.0 => 6.5.8 
    @react-navigation/elements: ^1.3.17 => 1.3.18 
    @react-navigation/native: ^6.0.8 => 6.1.7 
    @react-navigation/stack: ^6.2.0 => 6.3.17 
    @rnx-kit/align-deps: ^2.0.1 => 2.2.4 
    @shopify/flash-list: ^1.4.0 => 1.4.3 
    @trivago/prettier-plugin-sort-imports: ^4.0.0 => 4.1.1 
    @tsconfig/react-native: ^2.0.2 => 2.0.3 
    @types/jest: ^29.2.1 => 29.5.2 
    @types/lodash: ^4.14.191 => 4.14.195 
    @types/react: ^18.0.24 => 18.2.14 
    @types/react-native: ^0.70.4 => 0.70.14 (0.71.3, 0.70.11)
    @types/react-native-indicators: ^0.16.2 => 0.16.2 
    @types/react-native-vector-icons: ^6.4.13 => 6.4.13 
    @types/react-test-renderer: ^18.0.0 => 18.0.0 
    @typescript-eslint/eslint-plugin: ^5.37.0 => 5.59.11 (5.48.1)
    @typescript-eslint/parser: ^5.37.0 => 5.59.11 (5.48.1)
    HelloWorld:  0.0.1 
    amazon-cognito-identity-js: ^6.2.0 => 6.3.1 (6.2.0)
    amazon-cognito-identity-js/internals:  undefined ()
    apollo3-cache-persist: ^0.14.1 => 0.14.1 
    aws-amplify: ^5.2.7 => 5.3.1 
    babel-jest: ^29.2.1 => 29.5.0 
    babel-plugin-module-resolver: ^4.1.0 => 4.1.0 
    date-fns: ^2.29.3 => 2.30.0 
    deepmerge: ^4.3.1 => 4.3.1 (3.3.0)
    eslint: ^8.19.0 => 8.31.0 
    expo: ^48.0.17 => 48.0.19 
    expo-image-picker: ^14.1.1 => 14.1.1 
    graphql: ^16.6.0 => 16.7.1 (15.8.0)
    graphql-ws: ^5.12.1 => 5.14.0 (5.13.1)
    i18next: ^22.4.11 => 22.5.1 
    jest: ^29.2.1 => 29.5.0 
    lodash: ^4.17.21 => 4.17.21 
    metro-react-native-babel-preset: ^0.73.7 => 0.73.9 (0.73.10)
    patch-package: ^6.5.0 => 6.5.1 
    prettier: 2.8.3 => 2.8.3 
    react: 18.2.0 => 18.2.0 
    react-hook-form: ^7.44.3 => 7.45.0 
    react-i18next: ^12.2.0 => 12.3.1 
    react-native: ^0.71.0 => 0.71.11 
    react-native-apollo-devtools-client: ^1.0.4 => 1.0.4 
    react-native-blob-util: ^0.18.2 => 0.18.3 
    react-native-bootsplash: ^4.5.2 => 4.7.3 
    react-native-code-push: ^8.0.0 => 8.0.2 
    react-native-collapsible-tab-view: ^5.0.0-rc.13 => 5.0.0-rc.13 
    react-native-config: ^1.4.12 => 1.5.1 
    react-native-date-picker: ^4.2.13 => 4.2.13 
    react-native-fast-image: ^8.6.3 => 8.6.3 
    react-native-flipper: ^0.182.0 => 0.182.0 
    react-native-gesture-handler: ^2.9.0 => 2.12.0 
    react-native-get-random-values: ^1.8.0 => 1.9.0 
    react-native-image-crop-picker: ^0.40.0 => 0.40.0 
    react-native-indicators: ^0.17.0 => 0.17.0 
    react-native-keyboard-aware-scroll-view: ^0.9.5 => 0.9.5 
    react-native-linear-gradient: ^2.6.2 => 2.7.3 
    react-native-localize: ^3.0.0 => 3.0.1 
    react-native-modal: ^13.0.1 => 13.0.1 
    react-native-paper: ^5.8.0 => 5.8.0 
    react-native-permissions: ^3.8.0 => 3.8.1 
    react-native-reanimated: ^3.0.0 => 3.3.0 
    react-native-safe-area-context: ^4.6.3 => 4.6.3 
    react-native-screens: ^3.19.0 => 3.22.0 
    react-native-size-matters: ^0.4.0 => 0.4.0 
    react-native-svg: ^13.8.0 => 13.9.0 
    react-native-toast-message: ^2.1.6 => 2.1.6 
    react-native-vector-icons: ^9.2.0 => 9.2.0 
    react-test-renderer: 18.2.0 => 18.2.0 
    ts-node: ^10.9.1 => 10.9.1 
    typescript: ^5.1.3 => 5.1.3 
    yarn: ^1.22.19 => 1.22.19 
    yup: ^1.0.2 => 1.2.0 
  npmGlobalPackages:
    npm: 9.6.7
    yarn: 1.22.19

```

</details>

### Describe the bug

I recently update amplify to the latest version (5.3.1) and I noticed that files that have been uploaded to S3 bucket had different size (much smaller than the size on local). But after I downgrade my amplify version to 5.2.7 then the issue is no longer exists.

Image size when uploading using amplify version 5.2.7
<img width="1057" alt="Screenshot 2023-06-27 at 4 22 29 PM" src="https://github.com/aws-amplify/amplify-js/assets/80971410/ef475c47-a8c3-452a-a58a-96efabcea374">

Image size when uploading using amplify version 5.3.1
<img width="1055" alt="Screenshot 2023-06-27 at 4 23 24 PM" src="https://github.com/aws-amplify/amplify-js/assets/80971410/b26fbd37-6a69-4582-998a-22bfaf81a807">

### Expected behavior

Files size should be the same as the size before being uploaded.

### Reproduction steps

See the description above and code snippet below.

### Code Snippet

```javascript
// Put your code below this line.

// Configure Amplify
Amplify.configure({
    Auth: {
      identityPoolId: Config.AWS_IDENTITY_POOL_ID,
      region: Config.AWS_COGNITO_REGION,
    },
    Storage: {
      AWSS3: {
        bucket: Config.AWS_BUCKET_NAME,
        region: Config.AWS_BUCKET_REGION,
      },
    },
    ...configs,
  })

// Upload file
Storage.put(`${name}.${extension}`, file, {
      contentType: file.type,
      level: 'public',
      customPrefix: {
        public: '',
      },
    })
```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by nadetastic** [comment]
Hi @quocluongha, thank you for opening this issue. I have taken a look at this and from my testing Im not able to reproduce the issue. I have tested with `5.3.1` and the latest version which is `5.3.2` as of today on iOS simulator and I get the same file size uploaded.

Could you provide any additional information for this such as:
- Are you experiencing this on iOS or Android or both?
- Is this happening on Emulator/Simulator or on a physical device?
- Can you confirm if this is on Expo or bare React Native app?

**Comment by nadetastic** [comment]
@quocluongha Following up here - after additional testing I'm now able to reproduce this issue and it seems to be specific to Expo on the latest version (or at least since `5.3.1`). Will mark this as a bug and follow up soon.

Let me know if you have any additional questions or comments.

Thanks

**Comment by quocluongha** [comment]
Thanks @nadetastic. I'm sorry for late response. I used bare React Native with Expo Modules.

**Comment by nadetastic** [comment]
PR with fix https://github.com/aws-amplify/amplify-js/pull/11555

**Comment by jimblanc** [comment]
Fix has been released with `aws-amplify@5.3.3`, could you give it a shot with your app?

**Comment by quocluongha** [comment]
Thank you. It backs to normal again.
